src/storybook/story-test-runner/open-story/hermione-open-story.ts

```diff
@@ -0,0 +1,159 @@
+import type EventEmitter from "events";
+
+export type StoryLoadResult = {
+    rootSelector: string;
+    playFunctionError?: string;
+    loadError?: string;
+    notInjected?: true;
+};
+
+type HTMLElement = {
+    innerText: string;
+};
+
+type StorybookWindow = Window &
+    typeof globalThis & {
+        __HERMIONE_OPEN_STORY__: (storyId: string, remountOnly: boolean, done: (result: string) => void) => void;
+        __STORYBOOK_ADDONS_CHANNEL__: EventEmitter;
+        __STORYBOOK_ADDON_INTERACTIONS_INSTRUMENTER_STATE__: Record<
+            string,
+            {
+                calls: Array<{
+                    exception?: {
+                        message: string;
+                    };
+                }>;
+            }
+        >;
+    };
+
+export async function inject(browser: WebdriverIO.Browser): Promise<void> {
+    await browser.execute(`window.__HERMIONE_OPEN_STORY__ = ${openStoryScript.toString()}`);
+}
+
+export async function execute(
+    browser: WebdriverIO.Browser,
+    storyId: string,
+    remountOnly: boolean,
+): Promise<StoryLoadResult> {
+    const getResult = (): Promise<StoryLoadResult> =>
+        browser.executeAsync(executeOpenStoryScript, storyId, remountOnly).then(JSON.parse);
+
+    const result: StoryLoadResult = await getResult();
+
+    if (!result.notInjected) {
+        return result;
+    }
+
+    await inject(browser);
+
+    const retriedResult = await getResult();
+
+    if (retriedResult.notInjected) {
+        throw new Error("Can't inject client script");
+    }
+
+    return retriedResult;
+}
+
+export default { inject, execute };
+
+function openStoryScript(storyId: string, remountOnly: boolean, done: (result: string) => void): void {
+    function onPageLoad(fn: () => void): void {
+        if (document.readyState === "complete") {
+            fn();
+        } else {
+            document.onreadystatechange = function () {
+                if (document.readyState === "complete") {
+                    fn();
+                }
+            };
+        }
+    }
+
+    onPageLoad(function () {
+        const channel = (window as StorybookWindow).__STORYBOOK_ADDONS_CHANNEL__;
+        const result: StoryLoadResult = { rootSelector: "" };
+
+        function doneJson(value: StoryLoadResult): void {
+            channel.off("playFunctionThrewException", onPlayFunctionThrewException);
```

Comment:
Since this is browser context, we can't use it like `events.playFunctionThrewException`. (this `events` would need to be defined in browser context)